spec/integration/gds_sso_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.describe "usage of gds-sso gem" do
+  let(:omniauth_hash) do
+    OmniAuth::AuthHash.new({
+      provider: "gds",
+      uid: "123456",
+      info: {
+        email: "test@example.com",
+        name: "Test User",
+      },
+      extra: {
+        user: {
+          permissions: ["---"],
+          organisation_slug: "test-org",
+          organisation_content_id: "00000000-0000-0000-0000-000000000000",
+          disabled: false,
+        },
+      },
+    })
+  end
+
+  describe "authentication" do
+    before do
+      OmniAuth.config.test_mode = true
+      OmniAuth.config.mock_auth[:gds] = nil
+    end
+
+    after do
+      OmniAuth.config.test_mode = false
+    end
+
+    it "redirects to OmniAuth when no user is logged in" do
+      logout
+
+      get "/"
+
+      expect(response).to redirect_to("/auth/gds")
+    end
+
+    it "authenticates with OmniAuth and Warden" do
+      OmniAuth.config.mock_auth[:gds] = omniauth_hash
+
+      get "/auth/gds"
```

Comment:
Is this path correct? I dont see it mentioned when I run `rails routes`

Reply:
Yes, it's part of OmniAuth [[1](https://github.com/omniauth/omniauth#integrating-omniauth-into-your-application)]:

> To use OmniAuth, you need only to redirect users to `/auth/:provider`, where `:provider` is the name of the strategy (for example, `developer` or `twitter`). From there, OmniAuth will take over and take the user through the necessary steps to authenticate them with the chosen strategy.

Because it's handled by the OmniAuth middleware and not by the Rails router these routes don't show up in `rails routes` however.